modules/vba_documents/lib/vba_documents/multipart_parser.rb

```diff
@@ -166,9 +167,22 @@ def self.consume_body_tempfile(lines, separator)
           tf.rewind
           return tf, true
         else
+          # AWS appends a new line at the end of the pdf, we must remove it to maintain the original sha256 value
+          line.chomp! if line.encoding.name == 'ASCII-8BIT' && /%%EOF\n\r\n/.match?(line)
```

Comment:
If we are removing a line in the file, and then recording the sha256, but AWS is storing the file with the extra newline, then wouldn't our upstream (maybe downstream? Don't know those terms yet haha but whoever is next after us) comparisons not match? Should we be removing the newline from the file in AWS?

Reply:
It shouldn't be, but I did see some tweaks I want to make on base64 payloads. We remove the newline because s3 stores it with an appended newline, so what we really want is the sha BEFORE it hits aws. But I'm going to go ahead and mock GCIO and make sure the sha matches at the end of that workflow and add another unit test, just so we can have that extra warm and fuzzy